src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -136,6 +138,17 @@ public class KNNSettings {
         Setting.Property.Deprecated
     );
 
+    /**
+     * build_vector_datastructure - This parameter determines whether to build vector data structure for knn fields during indexing
+     * and merging.
+     */
+    public static final Setting<Boolean> INDEX_KNN_BUILD_VECTOR_DATASTRUCTURE = Setting.boolSetting(
```

Comment:
should we build this setting as int rather than bool, so that when we have threshold based graph build same setting can be used.

Reply:
we can min value as -1 and max 1 to start with. and in next time we can increase the max value to be very high